File: LATIHAN4DPBO2023-main/C++/SivitasAkademik.cpp
// Saya Raisyad Jullfikar NIM 2106238 mengerjakan soal Latihan 4 dalam mata kuliah Desain Pemrograman Berorientasi 
// Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

// Import Library
#include <string>
#include <iostream>

class SivitasAkademik : public Human {
private:
// Deklar Private attr
  string __asal_universitas;
  string __email_edu;

public:
  // Constructor
  SivitasAkademik(string NIK = "", string name = "", string jenis_kelamin = "", string asal_universitas = "", string email_edu = "") : Human(NIK, name, jenis_kelamin) {
    this->__asal_universitas = asal_universitas;
    this->__email_edu = email_edu;
  }

  // Getter dan Setter
  string get_auniv() { return this->__asal_universitas; } // Method untuk mengambil data Asal Univ
  void set_auniv(string asal_universitas) { this->__asal_universitas = asal_universitas; } // Method untuk menset atau mengubah atau menambah data Asal Univ

  string get_email() { return this->__email_edu; } // Method untuk mengambil data Email
  void set_email(string email_edu) { this->__email_edu = email_edu; } // Method untuk menset atau mengubah atau menambah data Email

  // Destructor
  ~SivitasAkademik() {}
};

File: LATIHAN2DPBO2023-main/Python/SivitasAkademik.py
# Saya Raisyad Jullfikar NIM 2106238 mengerjakan Latihan 2 
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
# keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

# Import class Human
from Human import *

# Class Sivitas Akademik yang merupakan anak dari class Human
# Merupakan class yang berisikan data - data akademika
# Seperti Asal Univ dan Email Pendidikan (email edu)

class SivitasAkademik(Human) :
  # Deklar private attr
  __asal_universitas = ""
  __email_edu = ""
  
  # CONSTRUCTOR
  # Constructor kosong yang mengisi valuenya dengan default value atau string kosong
  def __init__ (self, NIK, name, jenis_kelamin, asal_universitas = "", email_edu = "") :
    super().__init__(NIK,name,jenis_kelamin)
    # Fungsi super untuk memberikan akses dari kelas induknya yaitu Human
    self.__asal_universitas = asal_universitas
    self.__email_edu = email_edu
    
    
  # SETTER AND GETTER
     
  def get_auniv(self) : # Method untuk mengambil data Asal Univ
    return self.__asal_universitas
  def set_auniv(self, asal_universitas) : # Method untuk menset atau mengubah atau menambah data Asal Univ
    self.__asal_universitas = asal_universitas;
    
  def get_email(self) : # Method untuk mengambil data Email
    return self.__email_edu
  def set_email(self, email_edu) : # Method untuk menset atau mengubah atau menambah data Email
    self.__email_edu = email_edu;
